app/controllers/concerns/api/v2/lookup_keys_common_controller.rb

```diff
@@ -125,6 +127,19 @@ def return_if_smart_mismatch
     end
   end
 
+  def cast_default_value
+    obj = smart_variable_id? ? "smart_variable" : "smart_class_parameter"
+
+    cast_value(obj, :default_value)
+  end
+
+  def cast_value(obj = :override_value, value = :value)
+    return unless params[obj] && params[obj].key?(value) && !params[obj][value].is_a?(String)
+    value = params[obj][value]
+    params[obj][value] = value if value.is_a?(Hash)
```

Comment:
This doesn't make sense to assign here then do another assignment on the next line.